internal/handlers/handlers.go

```diff
@@ -1,64 +1,76 @@
 package handlers
 
 import (
-	"fmt"
+	"errors"
 	"github.com/go-chi/chi/v5"
 	"io"
 	"math/rand"
 	"net/http"
+	"time"
 )
 
-const letterAll = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const (
+	letterAll = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	basicURL  = "http://localhost:8080/"
+)
 
 var ListURL = make(map[string]string)
 
 func GetHandler(w http.ResponseWriter, r *http.Request) {
-	// этот обработчик принимает только запросы, отправленные методом GET
-	if r.Method != http.MethodGet {
-		http.Error(w, "Only GET requests are allowed!", http.StatusBadRequest)
-		return
-	}
 
 	id := chi.URLParam(r, "id")
+	//данные для теста
+	if id == "testurl" {
+		ListURL[id] = "http://yandex.ru"
 
-	w.Header().Set("Location", ListURL[id])
+	}
+	if _, ok := ListURL[id]; !ok {
+		err := errors.New("BadRequest")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
-	w.WriteHeader(307)
-	fmt.Println(ListURL[id])
 	if ListURL[id] == "" {
+		w.WriteHeader(http.StatusNotFound)
 		w.Write([]byte("Не найдено"))
 	} else {
+		w.Header().Set("Location", ListURL[id])
+		w.WriteHeader(http.StatusTemporaryRedirect)
 		w.Write([]byte(ListURL[id]))
 	}
 
 }
```

Comment:
так удобнее читать код.
```suggestion
	if ListURL[id] == "" {
		w.WriteHeader(http.StatusNotFound)
		w.Write([]byte("Не найдено"))
		return
	} 
	w.Header().Set("Location", ListURL[id])
	w.WriteHeader(http.StatusTemporaryRedirect)
	w.Write([]byte(ListURL[id]))


}
```
и не помню писал ли раньше, можно еще использовать функцию http.Redirect для последних 3 строк

Reply:
+